File: bazel_base/py2/bazel_base/bazel_base/bazel_base/MultiValue.py

import bazel_base.Value


class MultiValue(bazel_base.Value.Value):
    def stringify(self, spaces=''):
        _spaces = spaces + (' '*2)
        split = '\n'+_spaces
        res = '['+split
        res += split.join([c.stringify() for c in self.children])
        res += spaces + ']'
        return res

    def __init__(self):
        super(MultiValue, self).__init__()
        pass

File: bazel_base/cpp/bazel_base/bazel_base/src/MultiValue.cpp

#include "MultiValue.h"

namespace bazel_base {

void MultiValue::append(SingleValue *obj) const {
  if (obj == nullptr) {
    return;
  }
  obj->father = (Com *)(this);
  this->children.push_back(obj);
}

void MultiValue::stringify(ostream &out) const {
  out << "[";
  bool first = true;
  for (const auto &value : this->children) {
    if (!first)
      cout << ", ";
    first = false;
    value->stringify(out);
  }
  out << "]";
}

} // namespace bazel_base
